### DISCUSSION TITLE
论文模板如何在WinEdt软件上使用？

### DISCUSSION THREAD

**Discussion by forylest** [sjtug/SJTUThesis#893] [提问] [answered]
![图片](https://user-images.githubusercontent.com/99483155/220123867-eecef595-40bd-4bda-9ae8-e00868d5b490.png)
在WinEdt上找不到可以像texstudio一样可以自定义编译模式的地方，就算选择了XeTex的编译模式还是会出错。

**Comment by LogCreative** [comment, marked as answer]
将配置更改为

![image](https://user-images.githubusercontent.com/61653082/220142321-53f83b57-b0ba-482a-a99a-b16ae3e217f2.png)

然后使用 XeLaTeX 编译

![image](https://user-images.githubusercontent.com/61653082/220142493-1cbcb0c0-e23f-476b-89b8-4579ab1073f0.png)

**Reply by LogCreative** [reply]
当然 Switches: 中的 `-xelatex` 是可选的，已经在 .latexmkrc 中定义。

**Reply by forylest** [reply]
![图片](https://user-images.githubusercontent.com/99483155/220148729-20d4bb18-ae0b-4608-b84a-0d108d163086.png)
我按照你图中的写法重新设置了xetex的参数，结果还是会出现找不到cls类文件的错误，我甚至在texlive的根目录中也按照latexmk文件的路径把sjtuthesis的文件复制了过去。

**Reply by LogCreative** [reply]
前一个问题你需要检查你的更改是否通过按下 OK 按钮生效。后一个问题仅移入texmf目录是不够的，还需要在命令行输入 texhash 刷新文件名数据库。

**Reply by forylest** [reply]
明白了已成功解决，十分感谢。另外想请教一下这些经验是不是只有经历过才知道，经常遇到的一些编译问题在软件的手册里都没有

**Reply by LogCreative** [reply]
一些编译问题是 LaTeX 本身的，可以考虑通过一些参考资料学习：比如 [lshort](https://ctan.org/pkg/lshort-zh-cn)，这种编译问题可能会经常出现，推荐经常查阅相关资料；另一小部分的问题是关于编辑器的，可以考虑通过阅读编辑器自带的帮助文档来解决，运行环境问题一般解决就不太会再出现（除非运行环境发生了变动），所以这种问题一般不用太在意。
